src/joyride/sci.cljs

```diff
@@ -46,10 +76,19 @@
                            (symbol? namespace)
                            (source-script-by-ns namespace)
                            (string? namespace) ;; node built-in or npm library
-                           (if (= "vscode" namespace)
+                           (cond 
+                             (= "vscode" namespace)
                              (do (sci/add-class! @!ctx 'vscode vscode)
                                  (sci/add-import! @!ctx (symbol (str @sci/ns)) 'vscode (:as opts))
                                  {:handled true})
+
+                             (active-extension? namespace)
+                             (let [[module-name module] (extension-module namespace)]
+                               (sci/add-class! @!ctx (symbol namespace) module)
+                               (sci/add-import! @!ctx (symbol (str @sci/ns)) (symbol namespace) (:as opts))
+                               {:handled true})
```

Comment:
Can I jump in? I was able to make it work by changing these lines to:

```clojure
                             (let [[module-name module] (extension-module namespace)]
                               (handle-require! !ctx
                                                (symbol namespace)
                                                (symbol (str @sci/ns))
                                                namespace
                                                module
                                                opts))
```

And then adding this function:

```clojure
(defn- handle-require! [!ctx class-ns import-ns namespace module {:keys [refer as]}]
  (sci/add-class! @!ctx class-ns module)
  (when as
    (sci/add-import! @!ctx import-ns (symbol namespace) as))

  (when refer
    (doseq [sym refer]
      (let [prop (gobject/get module sym)
            sub-libname (str class-ns "$" prop)]
        (sci/add-class! @!ctx sub-libname prop)
        (sci/add-import! @!ctx import-ns sub-libname sym))))

  {:handled true})
```

Supposedly, this private function should be reusable for other kinds of JS imports :)

Reply:
Yes, that's similar to the async.md docs, but in those docs I also used `munge` on the libname. Probably not necessary, but still :)

Reply:
Thanks! Saw this now, after I've pushed a fix for this. I did something similar. Without the reusability, though. Maybe I should also extract it as a function...